### ISSUE TITLE
initContainer images are not remapped when assets: containerRegistry is set

### ISSUE THREAD

**Issue by joelittlejohn** [kubernetes/kops#5302]
1. **What `kops` version are you running? The command `kops version`, will display
 this information.**

    Version 1.8.0 (git-5099bc5)

2. **What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

    v1.8.6

3. **What cloud provider are you using?**

    AWS, Beijing region (cn-north-1)

4. **What commands did you run?  What is the simplest way to reproduce this issue?**

    After `kops edit cluster` I set the following in the cluster config:

        assets:
          containerRegistry: docker.mycompany.com:5000

    Now when I run `kops update cluster`, many docker images are uploaded to our internal docker registry.

5. **What happened after the commands executed?**

    Our k8s nodes are pulling many images from our internal docker registry, which is great, however I can see from the kubelet logs that nodes are still pulling `quay.io/coreos/flannel:v0.9.1-amd64` during startup, even though kops pushed `quay.io/coreos/flannel:v0.9.1-amd64` to `docker.mycompany.com:5000/coreos/flannel:v0.9.1-amd64`. 

    From the logs:

        kube_docker_client.go:330] Pulling image "quay.io/coreos/flannel:v0.9.1-amd64": "16a827ca53c8: Downloading [========>

    The problem with nodes in China using quay.io is that nodes sometimes take up to 30mins to start, if access to the internet outside China is slow at the time (which happens frequently in this region).

6. **What did you expect to happen?**

    I expected that the image `coreos/flannel:v0.9.1-amd64` would always be pulled from `docker.mycompany.com:5000/coreos/flannel:v0.9.1-amd64`.

7. **Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

    Full manifest info has been dumped here: https://pastebin.com/raw/DG5jLN3d

**Comment by joelittlejohn** [comment]
I've noticed that when I describe the flannel DaemonSet I see the following:

```
  Init Containers:
   install-cni:
    Image:	quay.io/coreos/flannel:v0.9.1-amd64
...
  Containers:
   kube-flannel:
    Image:	docker.mycompany.com:5000/quay.io-coreos-flannel:v0.9.1-amd64
```

So it looks like kops has not used my custom container registry for the init container image, and this is what's causing the nodes to pull from quay.io.

**Comment by joelittlejohn** [comment]
Looks like this is fixed by #5427.
